scipy/sparse/csgraph/_matching.pyx

```diff
@@ -59,6 +59,9 @@ def maximum_bipartite_matching(graph, perm_type='row'):
     matching existed. As such, code written against 1.4.0 will not necessarily
     work on older versions.
 
+    If multiple valid solutions are possible, output may vary with SciPy and
+    Python version.
```

Comment:
Does the output solution also vary based on the Python version?

I think that only the version of SciPy changes that. I think we need confirmation, but we might want to change this note if this is the case and use the [`::warning` directive to catch the eyes](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-warning):

```suggestion
    .. warning::
        If multiple valid solutions are possible, the output
        solution may vary from one SciPy version to another.
```

This suggestion also applies to the other docstrings.

Reply:
It has already been shown to vary with the Python version in gh-18494, I believe. It might also change due to the platform. So I may just stop the sentence at `output may vary.`

Reply:
FWIW, this kind of variation is reasonably common across a lot of `scipy` (especially `linalg`), so I wouldn't make too much of a fuss about it with `::warning`. A plain sentence gets the job done, IMO.